ManageIQ/manageiq-providers-openstack
# Add STF Event monitor backend

Service Telemetry Framework is a new way how Events can be provided by OpenStack. This support was added in OSP16.

Adding backend support for this technology.

## Technology notes
The STF is based on QPID Proton, Ruby bindings: https://github.com/apache/qpid-proton/tree/master/ruby/
That requires having qpid-proton-c package on RPM-based systems or similar for DEB-based. In ManageIQ, it is already used by NUAGE provider.

## Testing steps
1. Have >=OSP16.1 with enabled STF for CloudForms Events
2. Add OSP provider with STF Event source
3. After initial refresh, perform an action in OpenStack (like image upload, provision an Instance or create network)
4. Check that targeted refresh was triggered and events are visible on Provider - Timelines page.
In not clear situations refer to evm.log and search for "STF" keyword.

Related to: https://bugzilla.redhat.com/show_bug.cgi?id=1660842
